Samples/NetworkMediaStreamer/Readme.md

```diff
@@ -0,0 +1,298 @@
+    uint16_t socketPort,
+    bool bSecure,
+    IRTSPAuthProvider* pAuthProvider,
+    PCCERT_CONTEXT* aServerCerts,
+    size_t uCertCount,
+    IRTSPServerControl** ppRTSPServerControl);
+
+``` 
+### Arguments  
+|  |  |
+| ----------- | ----------- |
+| pStreamers | Input map of MediaSink objects as IMediaExtension interface and the corresponding url suffix. |
+| socketPort | Input socket port on which the Server wil listen for RTSP requests. |
+| bSecure | Input bool flag indicating if the server uses secure tcp connection |  
+| pAuthProvider | Input pointer to the IRTSPAuthProvider interface to the authentication provider to be used by the server |
+| aServerCerts | Input array of server certificate contexts to use for secure tcp tls. |
+| uCertCount | Input number of certificate contexts in the array |
+| ppRTSPServerControl | Output pointer to the [IRTSPServerControl](###IRTSPServerControl) interface to the RTSP server instance |
+---
+### Return  
+>HRESULT 
+: S_OK if succeeded. HRESULT error if failed. 
+---
+
+## Authentication provider
+
+The authentication provider implements the following interfaces 
+1. IRTSPAuthProvider: This interface is used by the RTSP server to authenticate incoming client connections
+2. IRTSPAuthProviderCredStore: This interface is used by the administrative module of the app to add and remove credentials etc.  
+A base/sample implementation of AuthProvider is included in the source. An instance to the authentication provider can be created using -
+---
+```
+ HRESULT GetAuthProviderInstance(
+    AuthType authType,
+    LPCWSTR pResourceName,
+    IRTSPAuthProvider** ppRTSPAuthProvider);
+``` 
+### Arguments  
+|  |  |
+| ----------- | ----------- |
+| authType | Input array of MFMediaType objects which describe each stream of the sink |
+| pResourceName | Input resource name to which the credentials are bound |
+| ppRTSPAuthProvider | Output pointer to the IRTSPAuthProvider interface to the auth provider instance |
+---
+### Return  
+>HRESULT 
+: S_OK if succeeded. HRESULT error if failed. 
+---
+
+## Interface definitions
+---
+### IRTSPAuthProvider
+```
+IRTSPAuthProvider : public ::IUnknown
+{
+    virtual STDMETHODIMP GetNewAuthSessionMessage(HSTRING* pAuthSessionMessage) = 0;
+    virtual STDMETHODIMP Authorize(LPCWSTR pAuthResp, LPCWSTR pAuthSesMsg, LPCWSTR pMethod) = 0;
+};
+```
+`IRTSPAuthProvider::GetNewAuthSessionMessage(HSTRING* pAuthSessionMessage)`  
+Generates nonce and Gets new authentication request message from Auth provider
+
+| |  |  |
+| ----------- | ----------- | --------- |
+| pAuthSessionMessage | Output pointer to the Authentication request/description message to be sent to client as per [RFC7616](https://tools.ietf.org/html/rfc7616#section-3.7). | e.g. `WWW-Authenticate: Digest realm="BeyondTheWall", nonce="6fc93a0604338c68070b75e49745baeab46c507155040000", algorithm=MD5, charset="UTF-8", stale=FALSE` |
+| return value | S_OK if succeeded. HRESULT error if failed. | 
+| 

```

Comment:
For some method prototypes, we use this documentation above. Consistency?

### Return  
>HRESULT 
: S_OK if succeeded. HRESULT error if failed.

Reply:
yes. that was the idea . The static factory methods have a different format and the interface definitions have a different one.

---
In reply to: [493738161](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r493738161) [](ancestors = 493738161)